dali/test/python/test_operator_cast.py

```diff
@@ -45,16 +40,18 @@ def iter_setup(self):
         data = self.iterator.next()
         self.feed_input(self.data, data, layout=self.layout)
 
-def check_cast_operator_float16(device, batch_size):
+def check_cast_operator_float16(device, batch_size, in_type, out_type):
     input_shape=(300, 400, 3)
-    eii1 = RandomDataIterator(batch_size, shape=input_shape)
-    eii2 = RandomDataIterator(batch_size, shape=input_shape)
+    eii1 = RandomlyShapedDataIterator(batch_size, max_shape=input_shape, dtype=in_type)
+    eii2 = RandomlyShapedDataIterator(batch_size, max_shape=input_shape, dtype=in_type)
     compare_pipelines(
-        CastPipeline(device, batch_size, iter(eii1), [types.FLOAT16, types.FLOAT]),
-        CastPipeline(device, batch_size, iter(eii2), [types.FLOAT]),
-        batch_size=batch_size, N_iterations=3)
+        CastPipeline(device, batch_size, iter(eii1), [types.FLOAT16, out_type]),
+        CastPipeline(device, batch_size, iter(eii2), [out_type]),
+        batch_size=batch_size, N_iterations=5)
```

Comment:
Here we go again back and forth with the number of iterations in test. I'm not saying it's wrong here, just putting a comment to pay attention to it, to double-check we in fact want this.

Reply:
I did it concisely to clear the output buffer twice.